cpp/src/arrow/compute/kernels/vector_selection_filter_internal.cc

```diff
@@ -924,6 +924,10 @@ Result<std::shared_ptr<RecordBatch>> FilterRecordBatch(const RecordBatch& batch,
     return Status::Invalid("Filter inputs must all be the same length");
   }
 
+  if (filter.is_chunked_array()) {
+    return Status::Invalid("Chunked filter not supported for RecordBatches.");
+  }
```

Comment:
We could instead quite easily implement the solution you initially implemented in Python, aka create a Table out of the RecordBatch and then call FilterTable?

After all, it already works for arrays:
```python
>>> a = pa.array([1,2,3,4,5])
>>> f = pa.chunked_array([[True,False],[False,True,True]])
>>> pc.filter(a, f)
<pyarrow.lib.ChunkedArray object at 0x7f91e6960360>
[
  [
    1
  ],
  [
    4,
    5
  ]
]
```

Reply:
Yeah, my thought exactly but got stuck with shared pointers and all that and I do not think I will get anywhere today :D Maybe tomorrow ...

Reply:
My attempt: https://github.com/apache/arrow/pull/40971/commits/b4c730f026e0e4d534da099a7b53a242e9f93717